examples/addLiquidity/addLiquidityWithPermit2Signature.ts

```diff
@@ -0,0 +1,172 @@
+/**
+ * Example showing how to add liquidity to a pool while approving tokens through Permit2 signature.
+ * (Runs against a local Anvil fork)
+ *
+ * Run with:
+ * pnpm example ./examples/addLiquidity/addLiquidityWithPermit2Signature.ts
+ */
+import {
+    Address,
+    createTestClient,
+    http,
+    parseEther,
+    publicActions,
+    walletActions,
+} from 'viem';
+import {
+    AddLiquidityInput,
+    AddLiquidityKind,
+    AddLiquidity,
+    BalancerApi,
+    ChainId,
+    PriceImpact,
+    Slippage,
+    TEST_API_ENDPOINT,
+    CHAINS,
+    Permit2Helper,
+} from '../../src';
+import { ANVIL_NETWORKS, startFork } from '../../test/anvil/anvil-global-setup';
+import { getSlot } from 'examples/lib/getSlot';
+import { makeForkTx } from 'examples/lib/makeForkTx';
+
+async function runAgainstFork() {
+    // User defined inputs
+    const { rpcUrl } = await startFork(ANVIL_NETWORKS.SEPOLIA);
+    const chainId = ChainId.SEPOLIA;
+    // 50BAL-50WETH
+    const pool = {
+        id: '0x5e54dad6c2504d63473f95d8025b763fd5b893c6',
+        address: '0x5e54dad6c2504d63473f95d8025b763fd5b893c6' as Address,
+    };
+    const amountsIn = [
+        {
+            rawAmount: parseEther('0.0001'),
+            decimals: 18,
+            address: '0x7b79995e5f793a07bc00c21412e50ecae098e7f9' as Address,
+        },
+        {
+            rawAmount: parseEther('0.0001'),
+            decimals: 18,
+            address: '0xb19382073c7a0addbb56ac6af1808fa49e377b75' as Address,
+        },
+    ];
+    const slippage = Slippage.fromPercentage('1'); // 1%
+
+    const client = createTestClient({
+        mode: 'anvil',
+        chain: CHAINS[chainId],
+        transport: http(rpcUrl),
+    })
+        .extend(publicActions)
+        .extend(walletActions);
+
+    const userAccount = (await client.getAddresses())[0];
+
+    // build add liquidity call with permit2 approvals
+    const call = await addLiquidityWithPermit2Signature({
+        client,
+        userAccount,
+        rpcUrl,
+        chainId,
+        amountsIn,
+        poolId: pool.id,
+        slippage,
+    });
+
+    // Skip Permit2 approval during fork setup so we can test approvals through signatures
+    const approveOnPermit2 = false;
+
+    // Make the tx against the local fork and print the result
+    await makeForkTx(
+        call,
+        {
+            rpcUrl,
+            chainId,
+            impersonateAccount: userAccount,
+            forkTokens: amountsIn.map((a) => ({
+                address: a.address,
+                slot: getSlot(chainId, a.address),
+                rawBalance: a.rawAmount,
+            })),
+            client,
+        },
+        [...amountsIn.map((a) => a.address), pool.address],
+        call.protocolVersion,
+        approveOnPermit2,
+    );
```

Comment:
Nice move adding the flag that prevents the default cascade of approvals by `makeForkTx` so that we can know the example is working properly!

I do have a larger suggestion for team to consider that I will bring up during standup:

`makeForkTx` may confuse less experienced devs who might be distracted by token slots or not fully grasp how the call object the SDK creates is used to send the tx. 

This may be a stretch that requires a fair bit of refactor, but perhaps rolling a simplified fork setup for examples could be an option where we use the default anvil account #0 to
1. wrap ETH into wETH
2. swap wETH into BAL
3. add liquidity to wETH/BAL pool
4. any necessary setup for boosted or nested joins / removes

From there, we should be able to focus examples on just the code needed for each operation and it would be easier to explain a simplified fork setup in some detail

Not suggesting this as part of this PR and not too attached to this idea, just a suggestion that could enable examples to be a little more straightforward

P.S. John had [this idea for examples](https://github.com/balancer/b-sdk/issues/339#issuecomment-2337592756) a while back. Not sure if it applies to add / remove, but wanted to raise awareness

Reply:
I'll keep as is for now, since we're evaluating different approach for examples on a separate PR.